Simple Teachable machine example

yining1023 opened the pull request:
This PR -
- adds `knn.clearClass(index)` and `knn.getClassExampleCount()` functions to KNN Image Classifier
- adds a Teachable machine example that outputs gifs, allows upload gifs, shows confidence perentage.

![demo](https://user-images.githubusercontent.com/8662372/33530670-2ffa237c-d851-11e7-95cc-2d7490289f16.gif)

shiffman commented:
This is amazing!!! I wonder, should we separate out "demos" (that are designed and a bit flashier) from "examples" (that are simple and just code skeletons)?

yining1023 commented:
Thanks!:P  Good idea! I can create a 'demo' folder tonight.

shiffman commented:
Also, we should add something onto the page to reference / credit [google creative lab and teachable machines](teachablemachine.withgoogle.com). Here's [the repo/license](https://github.com/googlecreativelab/teachable-machine/blob/master/LICENSE).  Tagging @alexanderchen for any feedback!

shiffman commented:
Finally @yining1023 there are some elements here that might be nice to expose in the simple KNNImage classifier example (e.g. confidence score) without the design treatment, so let's think about that too?

yining1023 commented:
Yes!
I will add the confidence scores, example counts and reset a class to the simple KNNImage classifier example.

yining1023 commented:
I updated the simple KNNImage classifier example and added credit to simple teachable machine example. also moved it to a demos folder.

![demo](https://user-images.githubusercontent.com/8662372/33702961-deb98fb6-daf3-11e7-811f-33fc80618c5c.gif)

<img width="1229" alt="screen shot 2017-12-07 at 01 07 42" src="https://user-images.githubusercontent.com/8662372/33702966-e3389096-daf3-11e7-89ec-64be034986ef.png">

shiffman commented on examples/KNNImage/index.html:
```diff
@@ -22,7 +22,15 @@
-    <button id="buttonB">Train B</button>
+    <button id="resetA">Reset A</button>
+    <p><span id="exampleA">0</span> Examples in A</p>
+    <p>Confidence in A is: <span id="confidenceA">0</span></p>
```
I'm going to move these to down in the guessing section I think?

shiffman commented on examples/KNNImage/sketch.js:
```diff
@@ -32,6 +32,19 @@ function setup() {
 
+  // Reset buttons
+  resetBtnA = select('#resetA');
+  resetBtnA.mousePressed(() => {
```
I'll probably adjust this to not have arrow syntax just to be super beginner friendly (arrow syntax a-ok in teachable machines demo though!)

shiffman commented:
Fantastic! Merging!

alexanderchen commented:
Hi everyone! Could we just credit "Google" instead of "Google Creative
Lab?" Thank you so much, this is such a cool demo!

On Sun, Dec 17, 2017, 12:14 PM Daniel Shiffman <notifications@github.com>
wrote:

> Merged #29 <https://github.com/ITPNYU/p5-deeplearn-js/pull/29>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ITPNYU/p5-deeplearn-js/pull/29#event-1391141682>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAhiGVEzcwZjtzvdbAKpR_xwGmDCOu8Zks5tBUv7gaJpZM4Qz7al>
> .
>

cvalenzuela commented:
Credits have been updated! @alexanderchen
